ftplugin/crystal.vim

```diff
@@ -59,7 +59,7 @@ nnoremap <Plug>(crystal-format) :<C-u>CrystalFormat<CR>
 
 augroup plugin-ft-crystal
     autocmd!
-    autocmd BufWritePre * if &ft ==# 'crystal' && g:crystal_auto_format | CrystalFormat | endif
+    autocmd BufWritePre <buffer> if &ft ==# 'crystal' && g:crystal_auto_format | CrystalFormat | endif
```

Comment:
I had the same thinking. I opted to leave it to avoid any surprises, but I agree that it's redundant 95%+ of the time.